### PR TITLE
Paginator optimization

### PR DESCRIPTION
**Pull request by stern-shawn**

### CODE REVIEW

File: client/app/components/NavBar/index.js [hunk 1]
```diff
@@ -9,8 +9,9 @@ import styles from './styles.scss';
 const NavBar = () => (
   <nav className={styles.bar}>
     <LinkButton label="Home" to="/" />
-    <LinkButton label="About" to="/about" />
+    <LinkButton label="Blog" to="/page/1" />
     <LinkButton label="Store" to="/store" />
+    <LinkButton label="About" to="/about" />
   </nav>
```
Comment (Globolobo) [inline client/app/components/NavBar/index.js:15]: you should make this into a config-able + map

File: client/app/components/Paginator/index.js [hunk 2]
```diff
@@ -8,11 +9,12 @@ import styles from './styles.scss';
...
+const Paginator = ({ currPage, numPages }) => {
   // Wrapper function to prevent out of range page requests
-  const getPostsSafe = (target) => {
-    if (target > 0 && target <= numPages) {
-      getPosts(target);
+  const getPostsSafe = (e, target) => {
+    if (!(target > 0 && target <= numPages)) {
+      console.warn(`User attempted to request invalid page number: ${target}`);
+      e.preventDefault();
     }
```
Comment (Globolobo) [inline client/app/components/Paginator/index.js]: you should move this if outside of the function. You shouldn't call an empty function

File: client/app/containers/App/epic.js [hunk 3]
```diff
@@ -1,15 +1,48 @@
...
+    });
+
+// Intelligently use state to determine if we need to fetch the requested page,
+// or can simply use cached content. Also provides support for if we change pages
+// using react-router navigation instead of actions.
+const fetchPageEpic = (action$, store) =>
   action$.ofType(LOCATION_CHANGE)
-    .map(() => closeDrawer());
+    .switchMap((action) => {
+      // If location change is to blog pages, parse out the page and fetch that data
```
Comment (Globolobo) [inline client/app/containers/App/epic.js:28]: switchMap ? fanceeeh.

File: client/app/containers/HomePage/index.js [hunk 4]
```diff
@@ -4,14 +4,14 @@
...
+import React, { PropTypes } from 'react';
 import Helmet from 'react-helmet';
 
 // Components
 import LandingHeader from 'components/LandingHeader';
 import Blog from 'containers/Blog';
 
-const HomePage = () => (
+const HomePage = ({ routeParams }) => (
   <article>
```
Comment (Globolobo) [inline client/app/containers/HomePage/index.js:15]: if you're only using pageId, why pass routeParams as well

File: client/app/containers/App/epic.js (hunk 3 above)
Comment (stern-shawn) [reply client/app/containers/App/epic.js:28]: Mhmm, looking at this it seemed like a good way to reduce having a bunch of possibly colliding async results: http://stackoverflow.com/a/42227335/7027882

File: client/app/containers/HomePage/index.js (hunk 4 above)
Comment (stern-shawn) [reply client/app/containers/HomePage/index.js:15]: routeParams is the object passed into the React element by ReactRouter, this is how I'm catching it, then using the pageId property later. Is there a more specific approach, like pulling the pageId key explicitly in the destructuring on this line?

File: client/app/components/NavBar/index.js (hunk 1 above)
Comment (stern-shawn) [reply client/app/components/NavBar/index.js:15]: riiiight

File: client/app/containers/HomePage/index.js (hunk 4 above)
Comment (stern-shawn) [reply client/app/containers/HomePage/index.js:15]: Ah, I see.

```Javascript
const HomePage = ({ routeParams: { pageId } }) => (
```

Relevant [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Pulling_fields_from_objects_passed_as_function_parameter)

**Pull request merged**